### PR TITLE
allow to attach ISO to VM when PS is disabled

### PR DESCRIPTION
**Pull request by MaJin1996**
before: attach ISO to VM will succeed when PS is maintaining and disabled(ISO has been in PS) 
 and will fail when PS is disabled(ISO isn't in PS). NFS,Local,SMP will check the ISO whether be deleted but
Ceph didn't check the ISO whether be deleted.

except: attach ISO to VM will succeed when PS is disabled(ISO has been in PS) 
and will fail when PS is maintaining and disabled(ISO isn't in PS)
NFS,Local,SMP,Ceph check the ISO whether be deleted.

add 2 test case, if run the 2 test case in one case , it will fail caused by disconnected backup storage.

for

https://github.com/zstackio/issues/issues/1959

### CODE REVIEW

File: plugin/ceph/src/main/java/org/zstack/storage/ceph/primary/CephPrimaryStorageBase.java [hunk 1]
```diff
@@ -648,6 +660,7 @@ public void setPoolNames(List<String> poolNames) {
     public static final String KVM_HA_CANCEL_SELF_FENCER = "/ha/ceph/cancelselffencer";
     public static final String GET_FACTS = "/ceph/primarystorage/facts";
     public static final String DELETE_IMAGE_CACHE = "/ceph/primarystorage/deleteimagecache";
+    public static final String CHECK_BITS_PATH = "ceph/primarystorage/snapshot/checkbits";
```
Comment (mingjian2049) [inline plugin/ceph/src/main/java/org/zstack/storage/ceph/primary/CephPrimaryStorageBase.java]: 前面加个/

File: plugin/ceph/src/main/java/org/zstack/storage/ceph/primary/CephPrimaryStorageBase.java [hunk 2]
```diff
@@ -1326,19 +1337,67 @@ public String getSyncSignature() {
...
-                        }
+                    SimpleQuery<ImageCacheVO> q = dbf.createQuery(ImageCacheVO.class);
+                    q.add(ImageCacheVO_.imageUuid, Op.EQ, image.getInventory().getUuid());
+                    q.add(ImageCacheVO_.primaryStorageUuid, Op.EQ, self.getUuid());
+                    ImageCacheVO cache = q.find();
+                    if (cache != null ){
+                        final CheckIsBitsExistingCmd cmd = new CheckIsBitsExistingCmd();
+                        cmd.setInstallPath(cache.getInstallUrl());
+                        httpCall(CHECK_BITS_PATH, cmd, CheckIsBitsExistingRsp.class, new ReturnValueCompletion<CheckIsBitsExistingRsp>(null) {
+                            @Override
```
Comment (zxwing) [inline plugin/ceph/src/main/java/org/zstack/storage/ceph/primary/CephPrimaryStorageBase.java]: ReturnValueCompletion<CheckIsBitsExistingRsp>(null)，这个地方应该传chain而不是Null

Comment (zxwing) [inline plugin/ceph/src/main/java/org/zstack/storage/ceph/primary/CephPrimaryStorageBase.java]: ReturnValueCompletion(null)，这个地方应该传chain和trigger，而不是Null

**Comment by zxwing** [comment]
需要integreation test

**Comment by MaJin1996** [comment]
本来这四个函数我是放在一个test case 里的，但是执行到最后会因为 Ping Backup Storage失败而导致BS失联，导致test失败，即使我加上PING_PATH的默认返回true的处理函数也一样。

**Comment by MaJin1996** [comment]
@ZStack-Robot 
加载ISO到VM上时，只可能在主存储enabled和disabled状态加载成功，在maitenance时必定失败
并且主存储disabled时，只能在此ISO之前挂载过此PS且文件未被删除时成功。

**Pull request closed without merging**